clumper/clump.py

```diff
@@ -66,45 +66,36 @@ def read_json(cls, path):
     @classmethod
     def read_csv(cls, path, delimiter=",", fieldnames=None, nrows=None):
         """
-        Reads in a csv file. Can also read files from url.
-
-        Parameters
-        ----------
-        path : filename or url
-        delimiter: must be a single character. `,` is the default.
-        fieldnames: If `None`, the first row of the csv is assumed as the header.
-        nrows: Number of rows to read in. Useful when reading large files. If `None`, all rows are read.
+        Reads in a csv file. Can also read files from url. The delimiter should be a single character; the default is `,`.
+        `nrows` limits the number of rows read in and can be useful when dealing with large files; by default, all the rows will be read.
+        You can also supply fieldnames, by default, the first row of the csv file is treated as the header row.
 
         Usage:
 
         ```python
         from clumper import Clumper
 
-        clump = Clumper.read_csv("tests/iris.csv")
-        assert len(clump) == 150
+        clump = Clumper.read_csv("tests/monopoly.csv")
+        assert len(clump) == 22
 
-        clump = Clumper.read_csv("tests/iris.csv", nrows = 30)
-        assert len(clump) == 30
+        clump = Clumper.read_csv("tests/monopoly.csv", nrows = 10)
+        assert len(clump) == 10
 
         clump = Clumper.read_csv("https://calmcode.io/datasets/monopoly.csv")
         assert len(clump) == 22
+
+        clump = Clumper.read_csv("tests/monopoly.csv", fieldnames=list("ABCDEFGHIJK"))
```

Comment:
How does `"ABCDEFGHIJK"` relate to the `monopoly.csv` dataset?

Reply:
It's not really clear to me what filenames does looking at this example.

Reply:
The argument description also does not say what happens when the fieldnames are passed.